### PR TITLE
Add squid service

### PR DESCRIPTION
**Pull request by klausenbusk**
**Description of your changes:** Add squid forward proxy for testing air-gapped functionality.

Checklist:

* [x] I have bumped the chart version
* [ ] I have documented my changes if this is needed
* [x] I have described my changes in the "description of your changes" field
* [ ] I have tested this change on a running cluster and the Argocd dashboard shows healthy and synced
* [ ] I have created the necessary tests in the chart, if they are possible/necessary
* [x] I have squashed commits if necessary

---

~Hmm, is access logging to stdout working?~

### CODE REVIEW

File: charts/squid/chart/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    {{- include "chart.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "chart.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/squid-config-cm.yaml") . | sha256sum }}
```
Comment (MadsHT) [inline charts/squid/chart/templates/deployment.yaml:15]: Should we have checksum for the secrets?

**Review by MadsHT — COMMENTED** [review]
Single question

File: charts/squid/chart/templates/deployment.yaml (hunk 1 above)
Comment (klausenbusk) [reply charts/squid/chart/templates/deployment.yaml:15]: I say yes.

**Review by MadsHT — APPROVED** [review]

**Review by MadsHT — APPROVED** [review]
LGTM

**Pull request merged**